src/proxy/http/canonicalize.rs

```diff
@@ -178,13 +178,25 @@ impl Future for Task {
     type Error = ();
 
     fn poll(&mut self) -> Poll<(), ()> {
+        // If the receiver has been dropped, stop watching for updates.
+        let tx_ready = match self.tx.poll_ready() {
+            Err(_) => return Ok(Async::Ready(())),
+            Ok(r) => r,
+        };
+
         loop {
             self.state = match self.state {
                 State::Init => {
                     let f = self.resolver.refine(self.original.name());
                     State::Pending(Timeout::new(f, self.timeout))
                 }
                 State::Pending(ref mut fut) => {
+                    // Only poll the resolution for updates when the receiver is
+                    // ready to receive an update.
+                    if tx_ready.is_not_ready() {
```

Comment:
I don't have the full loop in my head: is possible to get bad to here after having sent once? If so, this poll would be old...